routers/web/repo/issue_new.go

```diff
@@ -396,8 +396,17 @@ func NewIssuePost(ctx *context.Context) {
 
 	log.Trace("Issue created: %d/%d", repo.ID, issue.ID)
 	if ctx.FormString("redirect_after_creation") == "project" && projectID > 0 {
-		ctx.JSONRedirect(ctx.Repo.RepoLink + "/projects/" + strconv.FormatInt(projectID, 10))
-	} else {
-		ctx.JSONRedirect(issue.Link())
+		project, err := project_model.GetProjectByID(ctx, projectID)
+		if err == nil {
+			// to reduce load the content from database again, we just set the owner/repo here
+			if project.Type == project_model.TypeOrganization {
+				project.Owner = ctx.Repo.Owner
+			} else {
+				project.Repo = ctx.Repo.Repository
+			}
+			ctx.JSONRedirect(project.Link(ctx))
+			return
+		}
```

Comment:
I guess it's better to do 

```
if project.Type == project_model.TypeOrganization {
    ctx.JSONRedirect(fmt.Sprintf("%s/-/projects/%d", ctx.Repo.Owner.HomeLink(), projectID))
} else {
    ctx.JSONRedirect(fmt.Sprintf("%s/projects/%d", ctx.Repo.RepoLink, projectID)
}
```

It is much simpler and no need that fragile trick.

Reply:
a56ee15f3415490a88900f13cd11cbe9bda9ea6b